v3/pkg/application/linux_cgo.go

```diff
@@ -1570,11 +1578,22 @@ func runChooserDialog(window pointer, allowMultiple, createFolders, showHidden b
 					}
 					count++
 				}
-				close(selections)
-				C.gtk_widget_destroy((*C.GtkWidget)(unsafe.Pointer(fc)))
 			}
-		}()
-	})
+		case C.GTK_RESPONSE_CANCEL:
+			// Canceled
+		}
+	}
+	var response C.gint
+
+	if allowMultiple {
+		response = C.gtk_dialog_run((*C.GtkDialog)(fc))
+	} else {
+		InvokeAsync(func() {
+			response = C.gtk_dialog_run((*C.GtkDialog)(fc))
+		})
+	}
+	go handleReponse(response)
```

Comment:
This is a race condition! InvokeAsync is asynchronous so who knows what value response will have.  If you need to feed it to the goroutine please call it inside the asynchronous function. Otherwise it's always receiving an uninitialized value.

Reply:
Good catch had that originally and moved it out I'll get that pushed back to the async